### ISSUE TITLE
HilbertSeries HS: what is the difference between HS.getPeriod() vs HS.get_nr_coeff_quasipol() vs. HS.getHilbertQuasiPolynomial().size() ?

### ISSUE THREAD

**Issue by fingolfin** [Normaliz/Normaliz#250]
I am currently cleaning up some code in NormalizInterface, in the process, looked at the code for handling conversion of `HilbertQuasiPolynomial` and `WeightedEhrhartQuasiPolynomial` from C++ to GAP. What struck me was this difference in the code:
```C++
    vector< vector<mpz_class> > HQ = HS.getHilbertQuasiPolynomial();
    const size_t n = HS.getPeriod();
    ret = NEW_PLIST(T_PLIST, n+1);
```
versus
```C++
    vector< vector<mpz_class> > ehrhart_qp = int_data.getWeightedEhrhartQuasiPolynomial();
    const size_t n = ehrhart_qp.size();
    ret = NEW_PLIST(T_PLIST, n+1);
```

So once, the `getPeriod()` result of the (Hilbert) series is assume to be the "size" of the quasi polynomial; and once we just use, well, the actual size of the quasi polynomial.

And then I noticed that there is also `HS.get_nr_coeff_quasipol()`, which seems to be a third way to access the same data? Probably not, but then what is the difference?

To find out, I tried to see how `normaliz` prints the quasi polynomial, but that didn't really help, as I found this in `output.cpp`:
```C++
    long period = HS.getPeriod();
    if (period == 1 && (HS_Denom.size() == 0
                        || HS_Denom.begin()->first== (long) HS_Denom.size())) {
        out << HilbertOrEhrhart+ "polynomial:" << endl;
        out << HS.getHilbertQuasiPolynomial()[0];
...
    } else {
...
            out<<HilbertOrEhrhart+ "quasi-polynomial of period " << period << ":" << endl;
            if(HS.get_nr_coeff_quasipol()>=0){
                out << "only " << HS.get_nr_coeff_quasipol() << " highest coefficients computed" << endl;
                out << "their common period is " << HS.getHilbertQuasiPolynomial().size() << "" << endl;                        
            }
            Matrix<mpz_class> HQP(HS.getHilbertQuasiPolynomial());
            HQP.pretty_print(out,true);
...`
```
So this code uses `HS.getPeriod()`, but it also refers to `HS.getHilbertQuasiPolynomial().size()` as "the period".

What's going on here? My guess (now that I have written this): If `HS.getHilbertQuasiPolynomial()` has been computed, then `HS.getPeriod()` and `HS.getHilbertQuasiPolynomial().size()` are equal; but perhaps one can compute `HS.getPeriod()` w/o having computed the quasi polynomial, which is why two ways to get this value exist. If that is true, then I can safely adjust the conversion code in GAP to use `.size()` for both types of quasi polynomials.

Ah, but now grepping the examples and tests, I found this in `test/test-Int/Condorcet-q.ref`:
```
Hilbert quasi-polynomial of period 4:
only 2 highest coefficients computed
their common period is 2
 0:  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 15982652919 56262656
 1:  0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 15528493056 56262656
with common denominator = 6939597901822221635907747840000
```
So this suggests that `getPeriod` and `size` can disagree, and that instead `size() is (perhaps?) always equal to `HS.get_nr_coeff_quasipol()`. In that case, the existing code in NormalizInterface is actually buggy.

tl;dr: are any of the three named values guaranteed to be equal? If not, what are examples when they are not (for one case, it seems I already found one example in the test suite).

**Comment by w-bruns** [comment]
I will have a look. There are two "periods":
1) The true period of the quasipolynomial. Should be returned by getPeriod().
2) The common period of the computed coefficients (a divisor of 1), and often a proper divisor). Should be returned by size().

**Comment by w-bruns** [comment]
Can we close this issue?

**Comment by w-bruns** [comment]
Should be clear now.
